### PR TITLE
feat(hide-crds): Hide under the hood CRDs based on blacklisted annota…

### PR DESCRIPTION
**Pull request by abhinandan13jan**
Addresses: https://jira.coreos.com/browse/ODC-2096
Will be able to test and take share screenshots only after the backend updates the annotations.

### CODE REVIEW

File: frontend/public/components/catalog/catalog-page.tsx [hunk 1]
```diff
@@ -104,16 +104,23 @@ export class CatalogListPage extends React.Component<CatalogListPageProps, Catal
...
-    );
+    const items = [
+      ...clusterServiceClassItems,
+      ...imageStreamItems,
+      ...templateItems,
+      ...extensionItems,
+      ...projectTemplateItems,
+    ];
+
+    const filteredItems = _.filter(items, function(item) {
```
Comment (divyanshiGupta) [inline frontend/public/components/catalog/catalog-page.tsx]: Change the callback to an arrow function.

**Comment by divyanshiGupta** [comment]
@abhinandan13jan please resolve lint errors.

File: frontend/public/components/catalog/catalog-page.tsx (hunk 1 above)
Comment (spadgett) [reply frontend/public/components/catalog/catalog-page.tsx]:
```suggestion
    const filteredItems = _.filter(items, ['obj', 'metadata', 'annotations', 'apps.kubernetes.io/internal-object']);
```

**Comment by spadgett** [comment]
@ecordell had suggested we use `openshift.console.ui.visibility.<crdname>: hidden` in the CSV.

File: frontend/public/components/catalog/catalog-page.tsx (hunk 1 above)
Comment (abhinandan13jan) [reply frontend/public/components/catalog/catalog-page.tsx]: Updated

**Comment by abhinandan13jan** [comment]
@divyanshiGupta Fixed lint errors
@spadgett This is as per the findings and comments in the epic of  https://jira.coreos.com/browse/ODC-2096
cc: @christianvogt

File: frontend/public/components/catalog/catalog-page.tsx [hunk 2]
```diff
@@ -104,16 +104,22 @@ export class CatalogListPage extends React.Component<CatalogListPageProps, Catal
...
+      ...templateItems,
+      ...extensionItems,
+      ...projectTemplateItems,
+    ];
+
+    const filteredItems = _.reject(items, [
+      'obj',
+      'metadata',
+      'annotations',
+      'apps.kubernetes.io/internal-object',
```
Comment (shawn-hurley) [inline frontend/public/components/catalog/catalog-page.tsx]: We can not use the `apps.kubernetes.io` namespace without approval from upstream. I agree with Evan and Sam about changing this annotation

**Comment by christianvogt** [comment]
@sbose78 are you aware of `openshift.console.ui.visibility.<crdname>: hidden` in the CSV

**Comment by spadgett** [comment]
It was a proposal that we haven't yet implemented.

File: frontend/public/components/catalog/catalog-page.tsx (hunk 2 above)
Comment (sbose78) [reply frontend/public/components/catalog/catalog-page.tsx]: Right, we are going for `operators.operatorframework.io` for now as per https://github.com/operator-framework/operator-lifecycle-manager/pull/1097

**Comment by sbose78** [comment]
@spadgett Do you plan to implement it? I would say let's just stick with https://github.com/operator-framework/operator-lifecycle-manager/pull/1097

**Comment by abhinandan13jan** [comment]
The annotation for blacklisting is now updated to `operators.operatorframework.io/internal-object:true`

**Comment by sbose78** [comment]
> The annotation for blacklisting CRDs in UI is now updated to operators.operatorframework.io/internal-object:true

Looks good.

**Comment by spadgett** [comment]
@abhinandan13jan Can you give an example of how this looks in the ClusterServiceVersion YAML? I want to make the same change in the CSV details page.

**Comment by christianvogt** [comment]
@spadgett @abhinandan13jan seems like there's some confusion. This annotation is on the CRD. Although I just noticed that the catalog items avoid fetching the CRDs and instead construct fake objects using the CSV data to populate the catalog. Therefore the items which are being filtered are never the actual CRDs and therefore we will be unable to check for this annotation.

**Comment by spadgett** [comment]
Normal users can't list CRDs, so this approach won't work for developer catalog.

**Pull request merged**